qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1520A.py

```diff
@@ -73,12 +215,182 @@ def __init__(self, parent: 'KeysightB1500', name: Optional[str], slot_nr,
             fluctuations by changing the bias and so on."""))
 
+        self.add_submodule('cv_sweep', CVSweep(self, 'cv_sweep'))
+
+        self.add_parameter(name='sweep_mode',
+                           initial_value=constants.SweepMode.LINEAR,
+                           vals=vals.Enum(*list(constants.SweepMode)),
+                           parameter_class=GroupParameter)
+
+        self.add_parameter(name='sweep_start',
+                           initial_value=0,
+                           unit='V',
+                           parameter_class=GroupParameter)
+
+        self.add_parameter(name='sweep_end',
+                           initial_value=0,
+                           unit='V',
+                           parameter_class=GroupParameter)
+
+        self.add_parameter(name='sweep_steps',
+                           initial_value=self._sweep_steps,
+                           vals=vals.Ints(1, 1001),
+                           parameter_class=GroupParameter)
+
+        self.add_parameter(name='chan',
+                           initial_value=self.channels[0],
+                           parameter_class=GroupParameter)
+
+        self.set_sweep_steps = Group([self.chan,
+                                      self.sweep_mode,
+                                      self.sweep_start,
+                                      self.sweep_end,
+                                      self.sweep_steps],
+                                     set_cmd='WDCV '
+                                             '{chan},'
+                                             '{sweep_mode},'
+                                             '{sweep_start},'
+                                             '{sweep_end},'
+                                             '{sweep_steps}',
+                                     get_cmd=self._get_sweep_steps(),
+                                     get_parser=self._get_sweep_steps_parser)
+
+        self.add_parameter(name='adc_coef',
+                           initial_value=1,
+                           parameter_class=GroupParameter,
+                           vals=vals.Ints(1, 100))
+
+        self.add_parameter(name='adc_mode',
+                           initial_value=constants.ACT.Mode.PLC,
+                           parameter_class=GroupParameter,
+                           vals=vals.Enum(*list(constants.ACT.Mode)), )
+
+        self.adc_group = Group([self.adc_mode, self.adc_coef],
+                               set_cmd='ACT {adc_mode},{adc_coef}',
+                               get_cmd=self._get_adc_mode(),
+                               get_parser=self._get_adc_mode_parser)
+
+        self.add_parameter(name='ranging_mode',
+                           set_cmd=self._set_ranging_mode,
+                           get_cmd=None)
+
+        self.add_parameter(name='measurement_range_for_non_auto',
+                           set_cmd=self._set_measurement_range_for_non_auto,
+                           get_cmd=None)
```

Comment:
needs validator to allow constants AND `None` (thanks to how `rc` method is working)